File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        tokens_list = []
        for token in tokens:
            if token in ["+", "-", "*", "/"]:
                last, next_last = tokens_list.pop(-1), tokens_list.pop(-1)
                if token == "*":
                    tokens_list.append(last * next_last)
                elif token == "/":
                    tokens_list.append(int(float(next_last) / last))
                elif token == "+":
                    tokens_list.append(last + next_last)
                elif token == "-":
                    tokens_list.append(next_last - last)
            else:
                tokens_list.append(int(token))
        return tokens_list[0];

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.ts
function evalRPN(tokens: string[]): number {
    let stack: number[] = [];

    for (let token of tokens) {
        if (token === "+") {
            const b = stack.pop()!;
            const a = stack.pop()!;
            stack.push(a + b);
        } else if (token === "-") {
            const b = stack.pop()!;
            const a = stack.pop()!;
            stack.push(a - b);
        } else if (token === "*") {
            const b = stack.pop()!;
            const a = stack.pop()!;
            stack.push(a * b);
        } else if (token === "/") {
            const b = stack.pop()!;
            const a = stack.pop()!;
            stack.push(Math.trunc(a / b));
        } else {
            stack.push(Number(token));
        }
    }

    return stack.pop()!;
};